mmdata.c

```diff
@@ -923,39 +919,39 @@ nmbrString *nmbrCat(nmbrString *string1,...) /* String concatenation */
   numArgs--;    /* The last argument (0) is not a string */
 
   /* Find out the total string length needed */
-  j = 0;
-  for (i = 0; i < numArgs; i++) {
+  long j = 0;
+  for (int i = 0; i < numArgs; i++) {
     argLength[i]=nmbrLen(arg[i]);
-    j=j+argLength[i];
+    j += argLength[i];
   }
   /* Allocate the memory for it */
-  ptr=nmbrTempAlloc(j+1);
+  temp_nmbrString *ptr = nmbrTempAlloc(j+1);
   /* Move the strings into the newly allocated area */
   j = 0;
-  for (i = 0; i < numArgs; i++) {
-    nmbrCpy(ptr+j,arg[i]);
-    j=j+argLength[i];
+  for (int i = 0; i < numArgs; i++) {
+    nmbrCpy(ptr + j, arg[i]);
+    j += argLength[i];
   }
-  return (ptr);
+  return ptr;
 
 }
 
 
 
 /* Find out the length of a nmbrString */
-long nmbrLen(nmbrString *s)
+long nmbrLen(const nmbrString *s)
 {
   /* Assume it's been allocated with poolMalloc. */
-  return (((long)(((long *)s)[-1] - (long)(sizeof(nmbrString))))
+  return (((long)(((const long *)s)[-1] - (long)(sizeof(nmbrString))))
```

Comment:
I don't think, the cast to long of the nominator is necessary:
((((const long *)s)[-1] - (long)(sizeof(nmbrString)))
is sufficient and still looks complex.
same holds for line 954

Reply:
There are a lot of redundant casts like this. I think it was being compiled in some mode where the compiler would give a warning if you did not use a cast even for implicit numeric up-casts, but I don't think our current compiler settings enforce this. I think I want to clean this up in a separate PR. (The cast `const long *` from `const nmbrString *` is unnecessary since `nmbrString` is a typedef for `long`.) The casts here look like defense against promotion in the division; probably if we read the C spec it can be simplified a lot.